lib/maropost/api.rb

```diff
@@ -54,13 +54,23 @@ def update(contact)
       end
 
       def change_email(old_email, new_email)
-        contact = find(old_email)
+        contact = nil
+        old_email_contact = find(old_email)
+        new_email_contact = find(new_email)
 
-        if contact.present?
-          contact.email = new_email
-          contact = update(contact)
+        if old_email_contact.present? && new_email_contact.present?
+          new_email_contact = new_email_contact.merge_settings(old_email_contact)
+          contact = update(new_email_contact)
+          Maropost::DoNotMailList.create(old_email_contact)
```

Comment:
why do you need this? you have the update_do_not_email_list? do you need the scope? 

```
def update_do_not_mail_list(contact)
        if contact.allow_emails?
          DoNotMailList.delete(contact)
        else
          DoNotMailList.create(contact)
        end
      end
```

Reply:
Need to add the old contact to the do not mail list. We can't delete contacts in Maropost.